mmdet/models/backbones/dla.py

```diff
@@ -0,0 +1,640 @@
+                style=style)
+            self.tree2 = block(
+                out_channels,
+                out_channels,
+                1,
+                dilation=dilation,
+                conv_cfg=conv_cfg,
+                norm_cfg=norm_cfg,
+                with_cp=with_cp,
+                dcn=dcn,
+                plugins=plugins,
+                style=style)
+        else:
+            self.tree1 = Tree(
+                levels - 1,
+                block,
+                in_channels,
+                out_channels,
+                stride,
+                root_dim=0,
+                root_kernel_size=root_kernel_size,
+                dilation=dilation,
+                root_residual=root_residual,
+                conv_cfg=conv_cfg,
+                norm_cfg=norm_cfg,
+                with_cp=with_cp,
+                dcn=dcn,
+                plugins=plugins,
+                style=style)
+            self.tree2 = Tree(
+                levels - 1,
+                block,
+                out_channels,
+                out_channels,
+                root_dim=root_dim + out_channels,
+                root_kernel_size=root_kernel_size,
+                dilation=dilation,
+                root_residual=root_residual,
+                conv_cfg=conv_cfg,
+                norm_cfg=norm_cfg,
+                with_cp=with_cp,
+                dcn=dcn,
+                plugins=plugins,
+                style=style)
+
+        if levels == 1:
+            self.root = Root(
+                root_dim,
+                out_channels,
+                root_kernel_size,
+                root_residual,
+                conv_cfg=conv_cfg,
+                norm_cfg=norm_cfg)
+        self.level_root = level_root
+        self.root_dim = root_dim
+        self.downsample = None
+        self.project = None
+        self.levels = levels
+        if stride > 1:
+            self.downsample = nn.MaxPool2d(stride, stride=stride)
+        if in_channels != out_channels:
+            self.project = nn.Sequential(
+                build_conv_layer(
+                    conv_cfg,
+                    in_channels,
+                    out_channels,
+                    kernel_size=1,
+                    stride=1,
+                    bias=False),
+                build_norm_layer(norm_cfg, out_channels)[1])
+
+    def forward(self, x, residual=None, children=None):
+        children = [] if children is None else children
+        bottom = self.downsample(x) if self.downsample else x
+        residual = self.project(bottom) if self.project else bottom
+        if self.level_root:
+            children.append(bottom)
+        x1 = self.tree1(x, residual)
+        if self.levels == 1:
+            x2 = self.tree2(x1)
+            x = self.root(x2, x1, *children)
+        else:
+            children.append(x1)
+            x = self.tree2(x1, children=children)
+        return x
+
+
+@BACKBONES.register_module()
+class DLANet(nn.Module):
```

Comment:
Can DALNet inherit from ResNet to reduce some code in initialization?

Reply:
The 'arch_settings' between DLANet and ResNet is different, so inheriting from ResNet has some problems.